src/v/kafka/server/group.cc

```diff
@@ -3218,4 +3222,102 @@ std::ostream& operator<<(std::ostream& o, const group::offset_metadata& md) {
     return o;
 }
 
+bool group::subscribed(const model::topic& topic) const {
+    if (_subscriptions.has_value()) {
+        return _subscriptions.value().contains(topic);
+    }
+    // answer conservatively
```

Comment:
q: who uses this? Just trying to understand why being conservative is ok.

Reply:
it's an internal api that will be used by offset retention. generally offsets won't be GCd if there are consumers. so being conservative here means that if we can't definitively answer that a subscription isn't present, then default to answering yes to to make GC conservative.

Reply:
Ah makes sense.